src/main/java/org/javarosa/xpath/expr/XPathExpression.java

```diff
@@ -247,4 +247,9 @@ public void print (Object o) {
     public int hashCode () {
         return this.toString().hashCode();
     }
+
+    /**
+     * Returns true if this expression is not idempotent with respect to the current state of the form.
+     */
+    public abstract boolean isNotIdempotent();
```

Comment:
I personally find methods that are inherently negated difficult to reason about. Any reason not to make this `isIdempotent`? It would avoid the awkward double negation at https://github.com/getodk/javarosa/pull/713/files#diff-ff9982ce5c3c5b5119956e3bd9a20d8934090f9e6be0affad5c1598d66f72072R43

Reply:
Totally agree with this. One thing I realized looking at this again is that `isNotIdempotent()` is often "faster" to computer for non-idempotent functions as we don't need to traverse the whole expression - we just need to find something that isn't idempotent. I'll make the change anyway as I don't think that's a big enough deal to preserve what definitely feels like a worse API, but I wanted to make sure I'd brought that up.